homework/homework21/test_library_pytest.py

```diff
@@ -0,0 +1,134 @@
+"""This program covers the "Library" program with tests"""
+
+import pytest
+from logger_config import configure_logger
+from homework.homework11.library import Books, User
+
+# Name of the module used for logging
+module_name = __name__
+
+# Configure logger
+logger = configure_logger(module_name)
+
+
+class TestLibrary:
+    """Test class for testing Library class"""
+
+    @pytest.fixture(autouse=True)
+    def setup_teardown(self):
+        """Setup and teardown for each test"""
+        logger.info("Setting up test")
+        yield
+        logger.info("Tearing down test")
+
+    @pytest.fixture
+    def book(self):
+        """Fixture for creating a book object"""
+        return Books('The Lord of the Rings',
+                     'J.R.R. Tolkien',
+                     9785171358136,
+                     1120)
+
+    @pytest.fixture
+    def user1(self):
+        """Fixture for creating the first user"""
+        return User()
+
+    @pytest.fixture
+    def user2(self):
+        """Fixture for creating a second user"""
+        return User()
```

Comment:
there is no diff between those 2 fixtures, so use 1

```suggestion
    @pytest.fixture
    def user(self):
        """Fixture for creating a user"""
        return User()
```

Reply:
I don't quite agree here, I made 2 users in order to distinguish them in my tests, if I make one fixture then I won't understand which user is acting with the book. I can change it, but I'm not sure if it will be right for my tests. If it's a criticism then I can make the edits

Reply:
have you tried ?
```
    @pytest.fixture
    def user(self):
        """Fixture for creating a user"""
        return User()

def test_user(user, user):
      ...
```